### DISCUSSION TITLE
Clear all stores

### DISCUSSION THREAD

**Discussion by Lord-Y** [vuejs/pinia#693] [Help and Questions] [answered]
Hi there,

I'd like to know it there is a global way to clear all stores when a user is disconnecting?
I mean, if I have 10 stores, do I have to do this 10 times:

```
const store0 = useStore0()
store0.$reset()
const store1 = useStore1()
store1.$reset()
const storeXX = useStoreX()
storeXX.$reset()
```

**Comment by posva** [comment, marked as answer]
There is no way to do it directly as most of the time a user disconnecting doesn't mean all stores to get `$reset()` and some setup based stores cannot be `$reset()`.

You can however create a plugin that holds a reference to every store created within a map or array and expose a function that removes them:

```js
let stores = []
pinia.use(({ store }) => {
  stores.push(store)
})

function resetAllStores() {
  stores.forEach(store => {
    store.$reset()
  })
  stores = []
})
```

**Reply by Lord-Y** [reply]
Thx.

**Reply by robsonsobral** [reply]
**Be careful!**

To reset the array of stores at the end of the function will make it not to work a second time.

---

Thank you so much for your help, @posva .
